src/Utility/Utils.cs

```diff
@@ -0,0 +1,96 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.IO;
+using System.Management.Automation;
+using System.Text;
+
+namespace Microsoft.Azure.Functions.PowerShellWorker.Utility
+{
+    internal class Utils
+    {
+        /// <summary>
+        /// Helper method to do additional transformation on the input value based on the type constraints specified in the script.
+        /// </summary>
+        internal static object TransformInBindingValueAsNeeded(PSScriptParamInfo paramInfo, ReadOnlyBindingInfo bindingInfo, object value)
+        {
+            switch (bindingInfo.Type)
+            {
+                case "blob":
+                case "blobTrigger":
+                    // For blob input, the input data could be an array of bytes or a string. Documentation says
+                    // that a user can specify the 'dataType' to make the input data of the expected type. (see
+                    // https://docs.microsoft.com/en-us/azure/azure-functions/functions-triggers-bindings#binding-datatype-property)
+                    //
+                    // However, I found this setting is somewhat broken:
+                    //  1. For the 'blobTrigger' type in-binding, the input data is always byte array and the host doesn't
+                    //     respect the "dataType" setting in 'function.json'.
+                    //  2. For the 'blob' type in-binding, the input data type is 'String' by default. When specified, the
+                    //    'binary' and 'string' dataType are respected, but 'stream' dataType is not respected.

```

Comment:
So for them to leverage this for `blob` bindings to get a Stream object, they need to set the datatype to `binary` and put Stream in their run.ps1?

Reply:
Probably should open an issue on the host about `stream` not working...

I'm wondering if we should make up for that limitation and turn it into a Stream if specified in the function.json?

Reply:
> they need to set the datatype to binary and put Stream in their run.ps1

Unfortunately, yes, they have to set `"dataType" : "binary"` in `function.json`. Otherwise, the host will send the blob content as a string.
In case of `"dataType" : "stream"`, the data sent by Host contains nothing () ...

> should open an issue on the host about 'stream' not working...

Yes, probably we should. For `blobTrigger`, it's even worse -- `"dataType"` is completely ignored ...

> if we should make up for that limitation and turn it into a Stream if specified in the function.json?

We cannot. In case of `"dataType" : "stream"`, the data sent by Host contains nothing (`DataCase == None`)

Reply:
Issue opened in the Host repo: https://github.com/Azure/azure-functions-host/issues/3995